src/cast_core/src/CrudActor.cpp

```diff
@@ -1132,6 +1135,86 @@ std::string getDbName(const PhaseContext& phaseContext) {
     return phaseDb ? *phaseDb : *actorDb;
 }
 
+enum Units { NANOSECOND, MICROSECOND, MILLISECOND, SECOND, MINUTE, HOUR };
+// represent a random delay
+class Delay {
+public:
+    Delay(const Node& node, GeneratorArgs args)
+        : number_generator{doubleGenerator(node["^TimeSpec"]["value"], args)} {
+        auto unit_string = node["^TimeSpec"]["units"].maybe<std::string>().value_or("seconds");
```

Comment:
Genny is effectively always `camelCase`. The `snake_case` comes in only from the stl (e.g. `value_or` is a function on `std::optional`).